src/common.jl

```diff
@@ -1,194 +1,267 @@
+        _prob = prob
+    end
 
-      for t in d_discontinuities
-        tdir_t = tdir * t
-        tdir_t0 < tdir_t ≤ tdir_tf && push!(tstops_internal, tdir_t)
-      end
+    # DiffEqBase.solve(prob, _alg, args...; kwargs...)
+    integrator = DiffEqBase.__init(_prob, _alg, args...; kwargs...)
+    integrator.dt = stepsize(integrator.cache.uT, integrator.opts.abstol) # override handle_dt! setting of initial dt
+    DiffEqBase.solve!(integrator)
+    integrator.sol
+end
 
-      push!(tstops_internal, tdir_tf)
+# used in continuous callbacks and related methods to update Taylor expansions cache
+function update_jetcoeffs_cache!(u,f,p,cache::TaylorMethodCache)
+    @unpack tT, uT, duT, uauxT, parse_eqs = cache
+    for i in eachindex(u)
+        @inbounds uT[i][0] = u[i]
+        duT[i].coeffs .= zero(duT[i][0])
     end
+    __jetcoeffs!(Val(parse_eqs.x), f, tT, uT, duT, uauxT, p)
+    return nothing
+end
 
-    # saving time points
-    saveat_internal = BinaryMinHeap{tType}()
-    if typeof(saveat) <: Number
-      if (t0:saveat:tf)[end] == tf
-        for t in (t0 + saveat):saveat:tf
-          push!(saveat_internal, tdir * t)
-        end
-      else
-        for t in (t0 + saveat):saveat:(tf - saveat)
-          push!(saveat_internal, tdir * t)
+# This function was modified from OrdinaryDiffEq.jl; MIT-licensed
+# DiffEqBase.addsteps! overload for ::TaylorMethodCache to handle continuous
+# and vector callbacks with TaylorIntegration.jl via the common interface
+function DiffEqBase.addsteps!(k,t,uprev,u,dt,f,p,cache::TaylorMethodCache,always_calc_begin = false,allow_calc_end = true,force_calc_end = false)
+    if length(k)<2 || always_calc_begin
+        if typeof(cache) <: OrdinaryDiffEqMutableCache
+        rtmp = similar(u, eltype(eltype(k)))
+        f(rtmp,uprev,p,t)
+        copyat_or_push!(k,1,rtmp)
+        f(rtmp,u,p,t+dt)
+        copyat_or_push!(k,2,rtmp)
+        else
+        copyat_or_push!(k,1,f(uprev,p,t))
+        copyat_or_push!(k,2,f(u,p,t+dt))
         end
-      end
-    elseif !isempty(saveat)
-      for t in saveat
-        tdir_t = tdir * t
-        tdir_t0 < tdir_t < tdir_tf && push!(saveat_internal, tdir_t)
-      end
     end
+    update_jetcoeffs_cache!(u,f,p,cache)
+    nothing
+end
 
-    # discontinuities
-    d_discontinuities_internal = BinaryMinHeap{tType}()
-    sizehint!(d_discontinuities_internal.valtree, length(d_discontinuities))
-    for t in d_discontinuities
-      push!(d_discontinuities_internal, tdir * t)
-    end
+# This function was modified from TaylorSeries.jl; MIT-licensed
+# evaluate! overload to handle AbstractArray
+import TaylorSeries: evaluate!
+function evaluate!(x::AbstractArray{Taylor1{T},N}, δt::S,
+        x0::AbstractArray{T,N}) where {T<:Number, S<:Number, N}
 
-    tstops_internal, saveat_internal, d_discontinuities_internal
+    # @assert length(x) == length(x0)
+    @inbounds for i in eachindex(x, x0)
+        x0[i] = evaluate( x[i], δt )
+    end
+    nothing
 end
```

Comment:
Why not  adding it directly to TaylorSeries?

Reply:
This is the only point which I haven't addressed yet; I will open the corresponding PR in TaylorSeries, and it will perhaps require a new TaylorSeries release

Reply:
Update: just opened the corresponding PR. See: https://github.com/JuliaDiff/TaylorSeries.jl/pull/252